In [1]:
from model_candidates import StableLM_Gamma_7b_Half, ELYZA_Llama_7b_Half, GPT_NeoX_4b
from tqdm import tqdm
import pandas as pd
from time import time
import pickle

c:\Users\Andrew\envs\nasc2024\lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
N2 = pd.read_csv('./GrammarDataset/N2_grammar_cleaned.csv')
N3 = pd.read_csv('./GrammarDataset/N3_grammar_cleaned.csv')

In [3]:
ALL_QUESTIONS = pd.concat((N2, N3)).reset_index().drop(columns=['index'])
ALL_QUESTIONS

Question  \
0                          ___とたん、眠くなった。   
1    弟は、やっと見つけた就職先なのに、___のうちに、もう辞めてしまった。   
2                         この料理、熱いうちに___。   
3                            交通費は___一方だ。   
4                             ___としています。   
..                                   ...   
209            先生の説明を聞いている___だんだんわかってきた。   
210                         今朝___喉が痛かった。   
211     祖父は年を取って、最近___なかった。今日もかぎの場所を忘れた。   
212                   この店には高い品物___並んでいる。   
213             机の上に本がたくさん重ねてあって、今にも___。   

                                      Options  Answer  
0                 a.勉強が終わった b.勉強をした c.勉強をしていた       a  
1    a.仕事を覚えたか覚えないか b.働いているかいないか c.友達がいるかいないか       a  
2          a.おいしいですよ b.召し上がってください c.いい香りいがします       b  
3                a.値上がりする b.値上がりしている c.値上がりした       a  
4        a.間も無く夏が終わろう b.今日は雨が降ろう c.今年の冬は寒くなろう       a  
..                                        ...     ...  
209                        a.うちに b.たびに c.ところを       a  
210               a.起きたとき b.起きたついでに c.起きてるうちに       a  
211                   a.忘れかけに b.忘れにくく c.忘れっぽく       c  
212                    a.ぐらい b.なんか c.しか d.ばかり       d  
213     a.崩れているようだ b.崩そうだ c.崩れてしまった d.崩れたら大変だ       b  

[214 rows x 3 columns]

In [4]:
def evaluate_model(model, question_df):
    start = time()
    answers = []
    for index, row in tqdm(question_df.iterrows()):
        answer = model.generate_answer(row.Question, row.Options.split(' '))
        answers.append(answer)
    end = time()
    duration = end - start
    return answers, duration

### StableLM_Gamma_7b Evaluation

Specification: No sampling, output most probable next token

The assessment is to check if the LLM picks the correct answer. Explanation is manually investigated to check if it is rational.

In [5]:
stableLM_Gamma_7b = StableLM_Gamma_7b_Half()

Loading checkpoint shards: 100%|██████████| 3/3 [00:13<00:00,  4.64s/it]


In [6]:
stableLM_Gamma_7b_ans, stableLM_runtime = evaluate_model(stableLM_Gamma_7b, ALL_QUESTIONS)

1it [00:20, 20.25s/it]

解答：[b]

理由：「勉強


2it [00:37, 18.26s/it]

a.仕事を覚えたか覚


3it [00:53, 17.45s/it]

[c] いい香りいがします


4it [01:11, 17.44s/it]

解答：[b]

### 解説:


5it [01:24, 16.06s/it]

[b]今日は雨が降ろう


6it [01:37, 14.89s/it]

[b] ソファに座り


7it [01:54, 15.73s/it]

正解：'b.歩きつつ'

###


8it [02:12, 16.28s/it]

解答：'b.始まっている最中に'


9it [02:19, 13.34s/it]

[b]


10it [02:37, 15.05s/it]

解答：[b]

### 解説:


10it [02:50, 17.04s/it]


KeyboardInterrupt: 

In [ ]:
with open('./predictions/stablelm.pkl', 'wb') as f:
    pickle.dump([stableLM_Gamma_7b_ans, stableLM_runtime], f)

In [ ]:
del(stableLM_Gamma_7b)

Evaluation Result

| Model Name             | #Questions | #Correct | Total Inference Time | Average Inference Time |
|------------------------|------------|----------|----------------------|------------------------|
| StableLM_Gamma_7b_Half | 214        |          |                      |                        |
| ELYZA_Llama_7b_Half    | 214        |          |                      |                        |
| GPT_NeoX_4b            | 214        |          |                      |                        |